Sources/SnapshotTesting/Snapshotting/UIImage.swift

```diff
@@ -99,12 +101,19 @@ private func compare(_ old: UIImage, _ new: UIImage, precision: Float) -> Bool {
   guard let newerContext = context(for: newerCgImage, data: &newerBytes) else { return false }
   guard let newerData = newerContext.data else { return false }
   if memcmp(oldData, newerData, byteCount) == 0 { return true }
-  if precision >= 1 { return false }
+  if precision >= 1 && subpixelThreshold == 0 { return false }
   var differentPixelCount = 0
-  let threshold = 1 - precision
-  for byte in 0..<byteCount {
-    if oldBytes[byte] != newerBytes[byte] { differentPixelCount += 1 }
-    if Float(differentPixelCount) / Float(byteCount) > threshold { return false}
+  let threshold = Int(1.0 - precision * Float(byteCount))
```

Comment:
Let's say, the image is 10x10 pixels. `byteCount` is then `10 * 10 * 4 = 400` bytes.

With `precision: 1.0` the `threshold` is then `1.0 - 1.0 * 400 = 1 - 400 = -399`.

The check `if differentPixelCount >= threshold {` will always be `true`.

It should have extra parentheses like that:

```swift
let threshold = Int((1.0 - precision) * Float(byteCount))
```

In addition, I would propose using `round` like

```swift
let threshold = Int(round((1.0 - precision) * Float(byteCount)))
```

Reply:
Thanks @dmitri-zganiaiko, well spotted 👍

Resolved in 0efeef4.